src/System.Net.WebSockets.WebSocketProtocol/src/System/Net/WebSockets/ManagedWebSocketExtensions.cs

```diff
@@ -49,6 +49,32 @@ internal static Task<int> ReadAsync(this Stream stream, Memory<byte> destination
                 }
             }
         }
+
+        internal static ValueTask WriteAsync(this Stream stream, ReadOnlyMemory<byte> source, CancellationToken cancellationToken = default)
+        {
+            if (MemoryMarshal.TryGetArray(source, out ArraySegment<byte> array))
+            {
+                return new ValueTask(stream.WriteAsync(array.Array, array.Offset, array.Count, cancellationToken));
+            }
+            else
+            {
+                byte[] buffer = ArrayPool<byte>.Shared.Rent(source.Length);
+                source.Span.CopyTo(buffer);
+                return new ValueTask(FinishWriteAsync(stream.WriteAsync(buffer, 0, source.Length, cancellationToken), buffer));
```

Comment:
Is `buffer.Length == source.Length` here?
Why not use the RoM overload instead?

Reply:
As noted above, this code has to also run in a `netstandard2.0` assembly (consumed by ASP.NET Core in particular).

Reply:
> Is buffer.Length == source.Length here?

Not necessarily.  buffer.Length >= source.Length.  The ArrayPool could have returned a longer array.

> Why not use the RoM overload of WriteAsync instead?

This is the implementation of that "overload". See https://github.com/dotnet/corefx/pull/27497#discussion_r171582194.

Reply:
> Not necessarily. buffer.Length >= source.Length. The ArrayPool could have returned a longer array.

Got it.

> This is the implementation of that "overload".

Lol, that's right. I didn't connect the dots.